File: src/test/java/com/github/eendroroy/sdk/openweather/api/test/support/LoggingInterceptor.java
package com.github.eendroroy.sdk.openweather.api.test.support;

import com.github.eendroroy.sdk.openweather.interceptor.AbstractOWInterceptor;
import okhttp3.Request;
import okhttp3.Response;

/**
 * @author indrajit
 */

public class LoggingInterceptor extends AbstractOWInterceptor {

    @Override
    public Request interceptRequest(Request request) {
        System.out.println();
        System.out.println("Request  --> " + request.url().toString());
        System.out.println("Request  --> " + rawRequestBody(request));
        return request;
    }

    @Override
    public Response interceptResponse(Response response) {
        System.out.println();
        System.out.println("Response --> " + response.code());
        System.out.println("Response --> " + response.message());
        System.out.println("Response --> " + rawResponseBody(response));
        return response;
    }
}

File: src/test/kotlin/com/github/eendroroy/sdk/openweathermap/api/test/support/LoggingInterceptor.kt
package com.github.eendroroy.sdk.openweathermap.api.test.support

import com.github.eendroroy.sdk.openweathermap.interceptor.AbstractOWInterceptor
import okhttp3.Request
import okhttp3.Response

/**
 * @author indrajit
 */
class LoggingInterceptor : AbstractOWInterceptor() {
    override fun interceptRequest(request: Request): Request {
        println()
        println("Request  --> " + request.url().toString())
        println("Request  --> " + rawRequestBody(request))
        return request
    }

    override fun interceptResponse(response: Response): Response {
        println()
        println("Response --> " + response.code())
        println("Response --> " + response.message())
        println("Response --> " + rawResponseBody(response))
        return response
    }
}